corehq/apps/hqadmin/service_checks.py

```diff
@@ -133,7 +134,10 @@ def check_celery():
         return celery_status
     else:
         # Retry because of https://github.com/celery/celery/issues/4758
-        if not celery_worker_status.success:
+        num_retries = 4
+        for _ in range(num_retries):
+            if celery_worker_status.success:
+                break
             time.sleep(5)
             celery_worker_status = _check_celery_workers()
```

Comment:
Why check worker status here if the check will always fail once it gets into the `else` clause after

https://github.com/dimagi/commcare-hq/blob/f0926561b835ba4d9018c19bcb5b09addd906267/corehq/apps/hqadmin/service_checks.py#L133-L134

?

I admittedly did not read all of https://github.com/celery/celery/issues/4758, so maybe I missed something, but would it make sense to restructure this like
```py
celery_status = _check_celery()
stati = [celery_status]
if celery_status.success:
    # Retry because of https://github.com/celery/celery/issues/4758
    for _ in range(4):
        celery_worker_status = _check_celery_workers()
        if celery_worker_status.success:
            return celery_status
        time.sleep(5)
    stati.append(celery_worker_status)
return ServiceStatus(False, '\n'.join(status.msg for status in stati))
```

Reply:
Thanks for that catch - I messed up the implementation of the retrial.  This should fix it: ef3bc2c5bbebbb4bf222bae94165c949ca39b36a

Reply:
The idea is that ```_check_celery_workers()``` occasionally returns transient errors, so if it returns an error, we should pause and retry N times before returning an error in the check.